### PR TITLE
CCM-967: Added proxy check before test runs

### PR DESCRIPTION
**Pull request by ChewieCB**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: azure/templates/run-tests.yml [hunk 1]
```diff
@@ -36,6 +36,27 @@ steps:
...
+      code=$(echo $errorResponse | jq -r ".code")
+
+      if [[ ($status == "404" && $code == "CM_NOT_FOUND") || ($status == "401" && $code == "CM_DENIED") ]]; then
+          exit 0
+      else
+          exit 1
+      fi
+    workingDirectory: $(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)
+    displayName: Check proxy is deployed
+    retryCountOnTaskFailure: 3
```
Comment (PhilipSkinner) [inline azure/templates/run-tests.yml:58]: Will this wait between retries, if so how long does it wait?

I think this value may need to be higher to cater for when its taking some time for the proxy to have been stood up.

**Review by PhilipSkinner — CHANGES_REQUESTED** [review]

File: azure/templates/run-tests.yml (hunk 1 above)
Comment (ChewieCB) [reply azure/templates/run-tests.yml:58]: From the azure docs it looks like it has exponential backoff built-in, starting with a few seconds and increasing per retry. They don't give a specific timing value though, I'd expect it varies due to performance.

> The failing task retries in seconds. The wait time between each retry increases after each failed attempt.

Comment (PhilipSkinner) [reply azure/templates/run-tests.yml]: Excellent, that sounds ideal - I mean not about the azure docs not being clear but the functionality sounds perfect.

**Review by PhilipSkinner — APPROVED** [review]
Approved!

**Pull request merged**